src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,595 @@
+		}
+
+		public DynamicallyAccessedMemberTypes GetThisParameterAnnotation (MethodDefinition method)
+		{
+			if (_methods.TryGetValue (method, out var ann) && ann.ParameterAnnotations != null) {
+
+				foreach (var (ParamName, Annotation) in ann.ParameterAnnotations)
+					if (ParamName == "this")
+						return Annotation;
+			}
+
+			return DynamicallyAccessedMemberTypes.None;
+		}
+
+		static DynamicallyAccessedMemberTypes GetMemberTypesForDynamicallyAccessedMemberAttribute (ArrayBuilder<Attribute> attributes, LinkContext _context, string _xmlDocumentLocation)
+		{
+			foreach (var attribute in attributes.ToArray ()) {
+				if (attribute.attributeName == "System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembers" && attribute.arguments.Count == 1) {
+					foreach (var argument in attribute.arguments.ToArray ()) {
+						if (argument == string.Empty)
+							break;
+						try {
+							return (DynamicallyAccessedMemberTypes) Enum.Parse (typeof (DynamicallyAccessedMemberTypes), argument);
+						} catch (ArgumentException) {
+							_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not parse argument '{argument}' specified in '{_xmlDocumentLocation}' as a DynamicallyAccessedMemberTypes", 2020));
+						}
+					}
+				}
+			}
+			return DynamicallyAccessedMemberTypes.None;
+		}
+
+		void Initialize ()
+		{
+			XPathNavigator nav = _document.CreateNavigator ();
+
+			if (!nav.MoveToChild ("linker", string.Empty))
+				return;
+
+			try {
+				ProcessAssemblies (_context, nav.SelectChildren ("assembly", string.Empty));
+			} catch (Exception ex) when (!(ex is XmlResolutionException)) {
+				throw new XmlResolutionException (string.Format ("Failed to process XML description: {0}", _document), ex);
+			}
+		}
+
+		private void ProcessAssemblies (LinkContext context, XPathNodeIterator iterator)
+		{
+			while (iterator.MoveNext ()) {
+				AssemblyDefinition assembly = GetAssembly (context, GetAssemblyName (iterator.Current));
+
+				if (assembly == null) {
+					_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not resolve assembly {GetAssemblyName (iterator.Current).Name} specified in {_xmlDocumentLocation}", 2007));
+					continue;
+				}
+
+				if (!ShouldProcessSubstitutions (iterator.Current)) {
+					return;
+				}
+
+				ProcessTypes (assembly, iterator.Current.SelectChildren ("type", string.Empty));
+			}
+		}
+
+		ArrayBuilder<Attribute> ProcessAttributes (XPathNodeIterator iterator)
+		{
+			var attributes = new ArrayBuilder<Attribute> ();
+			while (iterator.MoveNext ()) {
+				string attributeName = GetFullName (iterator.Current);
+				ArrayBuilder<string> arguments = GetAttributeChilds (iterator.Current.SelectChildren ("argument", string.Empty));
+				ArrayBuilder<string> fields = GetAttributeChilds (iterator.Current.SelectChildren ("field", string.Empty));
+				ArrayBuilder<string> properties = GetAttributeChilds (iterator.Current.SelectChildren ("property", string.Empty));
+
```

Comment:
This is problematic. Basically only arguments are positional (that is we can rely on their order to identify which one is which). Fields and properties are not, so we would need to know both the name and the value for each of them. I think for now it would be enough to support just arguments. If we need more, we can add it later on.